centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx

```diff
@@ -66,33 +86,63 @@ const DataTableItem = forwardRef(
         </ActionArea>
         {hasActions && (
           <MuiCardActions>
+            <span />
             <span>
+              {onEditAccessRights && (
+                <IconButton
+                  ariaLabel={labels.labelShareWithContacts}
+                  title={labels.labelShareWithContacts}
+                  onClick={pipe(onEditAccessRights, closeMoreActions)}
+                >
+                  <ShareIcon fontSize="small" />
+                </IconButton>
+              )}
               <IconButton
-                aria-label="delete"
-                data-testid="delete"
-                icon={<DeleteIcon />}
-                size="small"
-                variant="ghost"
-                onClick={onDelete}
-              />
-            </span>
-            <span>
-              <IconButton
-                aria-label="edit access rights"
-                data-testid="edit-access-rights"
-                icon={<ShareIcon />}
-                size="small"
-                variant="primary"
-                onClick={() => onEditAccessRights?.()}
-              />
-              <IconButton
-                aria-label="edit"
-                data-testid="edit"
-                icon={<SettingsIcon />}
-                size="small"
-                variant="primary"
-                onClick={() => onEdit?.()}
-              />
+                ariaLabel={labels.labelMoreActions}
+                title={labels.labelMoreActions}
+                onClick={openMoreActions}
+              >
+                <MoreIcon />
+              </IconButton>
+              <Menu
+                anchorEl={moreActionsOpen}
+                open={Boolean(moreActionsOpen)}
+                onClose={closeMoreActions}
+              >
+                <ActionsList
+                  actions={[
+                    ...(onEdit
```

Comment:
as the logic becomes slightly more complex, it'd be a good idea to start extracting it out in my opinion

Reply:
Yes, you are right. However, this component was initially designed to serve the dashboards exclusively. Nonetheless, I have created a refactoring ticket with the purpose of integrating this component as part of the Listing component, and then adding a prop variant = 'List' | 'Grid'